src/Terrain/GameMap.java

```diff
@@ -22,32 +22,36 @@ public GameMap(CommandingOfficers.Commander[] COs, MapInfo mapInfo)
     map = new Location[mapWidth][mapHeight];
 
     // Build the map locations based on the MapInfo data.
-    for(int y = 0; y < mapHeight; ++y)
+    for( int y = 0; y < mapHeight; ++y )
     {
-      for(int x = 0; x < mapWidth; ++x)
+      for( int x = 0; x < mapWidth; ++x )
       {
+        // We set HQs to labs here because we don't yet support using labs as HQs
```

Comment:
Isn't this backwards? This is turning Labs into HQs.

Reply:
And as a broader topic, weren't we talking about making Labs a production facility for experimental units? If we change all Labs to HQs, then we can't have Labs at all.

Reply:
...yup, it's backwards. Whoops.
We could just turn labs into HQs in the import script, but I'd like to support AWBW's use for labs.

Reply:
The labs are meaningless for capture victory if there are HQs present, though it does allow you to mix teams who must hold all HQs and teams who only have labs. Another point is that Labs can start neutral and be captured, while HQs typically just become cities when captured. We *could* support alternate game modes where that's not the case, I suppose?
Labs do serve another function in AWBW: you can soft-ban units as "Lab Units" so that only players who own a lab can build them. This facilitates interesting game modes like the FIRST Stronghold map (a 3v3) where only one player on each side can build footsoldiers; the other two players are stuck with their starting footies until they get a lab (which is roughly halfway through the enemy fort)
More flexibility is ultimately nicer, but the way AWBW uses them IMO gives a nice compromise between consistent mechanics and flexibility for the map maker.